docs/install-service/setup-openshift-390.md

```diff
@@ -0,0 +1,396 @@
+- We want to make sure that mount point entry in `/etc/fstab` file is working
+  as expected.
+
+Now we do the actual OpenShift cluster installation. Start the cluster installation with this
+command. I prefer to time it so that I get a rough idea about how long the cluster setup
+took:
+
+```bash
+$ time ansible-playbook -i hosts /usr/share/ansible/openshift-ansible/playbooks/deploy_cluster.yml -vvv
+```
+
+Once the cluster is setup, we need to bring up a Jenkins server using the jenkins
+persistent template available in the `openshift` namespace. But we need to perform a
+few steps before that. We need to:
+- Create a user that’s able to use the OpenShift web console (on openshift master):
+  ```bash
+  $ oc login -u system:admin --config ~/.kube/config
+  $ oc create user cccp
+  $ htpasswd /etc/origin/master/htpasswd cccp
+  ```
+
+- Create a project:
+  ```bash
+  # create cccp project using cccp user
+  $ oc login -u cccp
+  $ oc new-project cccp
+  ```
+
+- Create a PV using the 5GB NFS share we created earlier. This can be done only
+  as a cluster-admin user. Replace NFS VM IP address to correct one in
+  following yml template:
+  ```bash
+  $ cat jenkins-pv.yml
+  apiVersion: v1
+  kind: PersistentVolume
+  metadata:
+  name: jenkins
+  namespace: cccp
+  spec:
+  capacity:
+  storage: 5Gi
+  accessModes:
+  - ReadWriteOnce
+  nfs:
+  path: /jenkins
+  server: 172.29.33.8
+  persistentVolumeReclaimPolicy: Recycle
+
+  $ oc login -u system:admin --config ~/.kube/config
+
+  $ oc create -f jenkins-pv.yml
+  ```
+- Bring up a Jenkins server using the jenkins persistent template. This has to
+  be done as normal user and not as admin user. ​ Make sure that the
+  Jenkins PV you created in above step is formated/emptied. ​ I spent about
+  2 days trying to debug issues arising out of stale data on that NFS storage.
+  Also add an extra environment variable to make sure that parallel builds work
+  as expected:
+
+  ```bash
+  $ oc login -u cccp
+  $ oc process -p MEMORY_LIMIT=1Gi openshift//jenkins-persistent | oc create -f -
+
+  # add env var for parallel builds
+  $ oc set env dc/jenkins JENKINS_JAVA_OVERRIDES="-Dhudson.slaves.NodeProvisioner.initialDelay=0,-Dhudson.slaves.NodeProvisioner.MARGIN=50,-Dhudson.slaves.NodeProvisioner.MARGIN0=0.85"
+
+  # add required permissions
+  $ oc login -u system:admin
+  $ oc adm policy add-scc-to-user privileged system:serviceaccount:cccp:jenkins
+  $ oc adm policy add-role-to-user system:image-builder system:serviceaccount:cccp:jenkins
+
+  ```
+
+Now go to the OpenShift web console and check if Jenkins deployment has come up
+and if the PVC of that deployment is bound to the PV you created earlier.
+
+To access Jenkins web console, you’ll need to check the OpenShift web console for the
+route that was created with nip.io subdomain. This should be visible on the
+project/namespace’s homepage on OpenShift console.
+
+Configuring DaemonSet: Scanning is one of the build pipeline phase the service
```

Comment:
I think this deserves a heading of its own. But a sub heading under "Setting up the cluster".